tests/pytorch/test_dataloader.py

```diff
@@ -247,6 +247,103 @@ def test_node_dataloader(idtype, sampler_name, mode, use_ddp):
     if use_ddp:
         dist.destroy_process_group()
 
+@pytest.mark.parametrize('mode', ['cpu', 'uva', 'pure_gpu'])
+def test_external_feature_storage(mode):
+    if mode != "cpu" and F.ctx() == F.cpu():
+        pytest.skip("UVA and GPU sampling require a GPU.")
+
+    def _random_tensor(num):
+        t = torch.randn(num, 8)
+        if mode == "uva":
+            t = t.pin_memory()
+        elif mode == "pure_gpu":
+            t = t.to(F.cuda())
+        return t
+
+    # 1. homogenous graph
+    g1 = dgl.graph(([0, 0, 0, 1, 1], [1, 2, 3, 3, 4]))
+    g1.ndata["feat"] = F.copy_to(F.randn((5, 8)), F.cpu())
+    g1.ndata["label"] = F.copy_to(F.randn((g1.num_nodes(),)), F.cpu())
+    indices = F.copy_to(F.arange(0, g1.num_nodes()), F.cpu())
+    if mode == "pure_gpu":
+        g1 = g1.to(F.cuda())
+        indices = F.copy_to(indices, F.cuda())
+
+    sampler = dgl.dataloading.MultiLayerNeighborSampler(
+        [3, 3], prefetch_node_feats=["feat", "feat2"], prefetch_edge_feats=["e"]
```

Comment:
Does this still work if `feat2` and `e` are not prefetched?

Reply:
Not. It requires:
1. Specify the nodes/edges features to be fetched in `prefetch_node_feats`/`prefetch_edge_feats`, and
2. Attach them to the dataloader via `attach_ndata` and `attach_edata`.

Reply:
Is there a way we could prefetch these features by default?